test/run_test.py

```diff
@@ -677,6 +704,60 @@ def run_doctests(test_module, test_directory, options):
     return result
 
 
+def print_log_file(test, file_path):
+    with open(file_path, "r") as f:
+        print_to_stderr("")
+        print_to_stderr(f'PRINT LOG FILE of {test} ({file_path})')
+        print_to_stderr(f'##[group]PRINT LOG FILE of {test} ({file_path})')
+        print_to_stderr(f.read())
+        print_to_stderr('##[endgroup]')
+        print_to_stderr(f"FINISHED PRINT LOG FILE of {test} ({file_path})")
+        print_to_stderr("")
+    os.remove(file_path)
+
+
+def run_test_ops(test_module, test_directory, options):
+    if 'slow-gradcheck' in os.getenv("BUILD_ENVIRONMENT", ""):
+        # there are a lot of tests that take up a lot of space in slowgrad check, so don't bother parallelizing
+        # it's also on periodic so we don't care about TTS as much
+        return run_test(test_module, test_directory, copy.deepcopy(options),
+                        extra_unittest_args=["--use-pytest", '-vv', '-x', '--reruns=2', '-rfEX'],
+                        )
+
+    file_names = []
+    return_codes = []
+    num_procs = 3
+    os.environ["PARALLEL_TESTING"] = "1"
+    pool = mp.Pool(num_procs)
+    for i in range(num_procs):
+        log_fd, file_path = tempfile.mkstemp(dir=REPO_ROOT / "test" / "test-reports",
```

Comment:
I think creating the temp file here looks hard to maintain. Do you think it's better to do this internally and transparently inside run_test?  After the test, you can just glob all files under `REPO_ROOT/test/test-reports` as they are already named correctly with the test module prefix.  So there is no need to pass the temp filename and file descriptor around.  You can also just remove the whole test-reports directory after

Reply:
moved file creation + printing to inside of run_test